### ISSUE TITLE
Compile failure with glog 0.3

### ISSUE THREAD

**Issue by sandwichmaker** [ceres-solver/ceres-solver#16]
Original [issue 15](https://code.google.com/p/ceres-solver/issues/detail?id=15) created by sandwichmaker on 2012-05-05T21:47:18.000Z:

TL;DR: 

Doesn't compile with glog version 0.3 (unfortunately, the shipped version in Fedora 16).

Up to you what you do with this:
1. Can close it as &quot;won't fix&quot;: Fedora is shipping a seriously old glog, not your problem.
2. Explicitly check for a glog version &gt;= 0.3.1 which has the fix in it.
3. Add a doc note.

Details:

<b>What steps will reproduce the problem?</b>
1. Fedora 16
2. system supplied glog-devel (0.3), eigen3-devel, gflags-devel 

Get a compile error:

[  2%] Building CXX object internal/ceres/CMakeFiles/ceres.dir/block_jacobian_writer.o
/home/michael/projects/ceres-solver-1.0.0/internal/ceres/block_jacobian_writer.cc: In constructor ‘ceres::internal::BlockJacobianWriter::BlockJacobianWriter(const ceres::internal::Evaluator::Options&amp;, ceres::internal::Program_)’:
/home/michael/projects/ceres-solver-1.0.0/internal/ceres/block_jacobian_writer.cc:126:113: error: ‘GetReferenceableValue’ was not declared in this scope
/home/michael/projects/ceres-solver-1.0.0/internal/ceres/block_jacobian_writer.cc:126:113: note: suggested alternative:
/usr/include/glog/logging.h:578:27: note:   ‘google::GetReferenceableValue’
/home/michael/projects/ceres-solver-1.0.0/internal/ceres/block_jacobian_writer.cc: In member function ‘ceres::internal::SparseMatrix_ ceres::internal::BlockJacobianWriter::CreateJacobian() const’:
/home/michael/projects/ceres-solver-1.0.0/internal/ceres/block_jacobian_writer.cc:155:115: error: ‘GetReferenceableValue’ was not declared in this scope
/home/michael/projects/ceres-solver-1.0.0/internal/ceres/block_jacobian_writer.cc:155:115: note: suggested alternative:
/usr/include/glog/logging.h:578:27: note:   ‘google::GetReferenceableValue’

Looks like a namespace issue in glog?

Hmm. Looks like glog is busted:

// In optimized mode, use CheckOpString to hint to compiler that
// the while condition is unlikely.
# define CHECK_OP_LOG(name, op, val1, val2, log) \

  while (google::CheckOpString _result = \
         google::Check##name##Impl(GetReferenceableValue(val1), \
                           GetReferenceableValue(val2), \
                           #val1 &quot; &quot; #op &quot; &quot; #val2)) \
    log(__FILE__, **LINE**, _result).stream()
# endif  // STATIC_ANALYSIS, !NDEBUG

It's missing the google:: namespace prefix on GetReferenceableValue().

Hmm. For which a fix was checked into glog on Nov 11, 2009!
